tests/Core/Ruleset/ProcessRuleShouldProcessElementTest.php

```diff
@@ -0,0 +1,660 @@
+
+        // Verify that the non-selective property directive IS applied.
+        $propertyName = 'exact';
+        $expected     = true;
+
+        $this->assertXObjectHasProperty($propertyName, $sniffObject);
+        $this->assertSame($expected, $sniffObject->$propertyName, 'Non-selective property change directive not applied');
+
+        // Verify that the CS-only property directive is NOT applied.
+        $propertyName    = 'indent';
+        $expectedDefault = 4;
+
+        $this->assertXObjectHasProperty($propertyName, $sniffObject);
+        $this->assertSame($expectedDefault, $sniffObject->$propertyName, 'cs-only property change directive was applied');
+
+        // Verify that the CBF-only property directive IS applied.
+        $propertyName = 'tabIndent';
+        $expected     = true;
+
+        $this->assertXObjectHasProperty($propertyName, $sniffObject);
+        $this->assertSame($expected, $sniffObject->$propertyName, 'cbf-only property change directive not applied');
+
+    }//end testShouldProcessPropertyCbfonly()
+
+
+    /**
+     * Verify that in CS mode, phpcs-only <element> directives are set and phpcbf-only <element>
+     * directives are ignored.
+     *
+     * @return void
+     */
+    public function testShouldProcessElementCsonly()
+    {
+        if (PHP_CODESNIFFER_CBF === true) {
+            $this->markTestSkipped('This test needs CS mode to run');
+        }
+
+        $expected = [
+            // Non-selective element directive.
+            'T_COMMENT',
+            // Phpcs-only element directive.
+            'T_CLASS',
+            // Non-selective element directive via `extend`.
+            'T_BACKTICK',
+            // Phpcs-only element directive via `extend`.
+            'T_INTERFACE',
+        ];
+
+        $this->verifyShouldProcessElement($expected);
+
+    }//end testShouldProcessElementCsonly()
+
+
+    /**
+     * Verify that in CBF mode, phpcbf-only <element> directives are set and phpcs-only <element>
+     * directives are ignored.
+     *
+     * @group CBF
+     *
+     * @return void
+     */
+    public function testShouldProcessElementCbfonly()
+    {
+        if (PHP_CODESNIFFER_CBF === false) {
+            $this->markTestSkipped('This test needs CBF mode to run');
+        }
+
+        $expected = [
+            // Non-selective element directive.
+            'T_COMMENT',
+            // Phpcbf-only element directive.
+            'T_ENUM',
+            // Non-selective element directive via `extend`.
+            'T_BACKTICK',
+            // Phpcbf-only element directive via `extend`.
+            'T_TRAIT',
+        ];
+
+        $this->verifyShouldProcessElement($expected);
+
+    }//end testShouldProcessElementCbfonly()
+
+
+    /**
+     * Verify that <element> directives are set correctly.
+     *
+     * @param array<string> $expected Expected sniff property value.
+     *
+     * @return void
+     */
+    public function verifyShouldProcessElement($expected)
```

Comment:
Should this be a `private` method as I believe it is a helper method used only in this class?

Reply:
Fair point.